### PR TITLE
Helpers for spells

### PR DESCRIPTION
**Pull request by ShiftyRail**
Helpers for spells, fix the hardcoding issues for #16819

Also did absent-mindedly a bunch of absolute pathing refactors in the dna folders, I'll revert it if I'm asked.

Also fully tested, I get the exact same list before the PR and after it when I open a spellbook.

### CODE REVIEW

File: code/game/gamemodes/wizard/spellbook.dm [hunk 1]
```diff
@@ -9,44 +9,9 @@
 	throw_range = 5
 	w_class = W_CLASS_TINY
 	flags = FPRINT
+	var/debug_nb_sorts = 0
```
Comment (ShiftyRail) [inline code/game/gamemodes/wizard/spellbook.dm]: DEBUG VARIABLE AA

**Review by ShiftyRail — COMMENTED** [review]
>cannot request changes on your own PR

**Comment by ShiftyRail** [comment]
Should be finished.

**Comment by ShiftyRail** [comment]
This time it should be all good, tested and all.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Use defines for all the spell types.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**